chrome/extension/background/icon.js

```diff
@@ -11,31 +11,39 @@ import icon16Saved from './../../assets/img/icon-16-saved.png';
 import icon32Saved from './../../assets/img/icon-32-saved.png';
 import icon128Saved from './../../assets/img/icon-128-saved.png';
 
+const setBadge = (tabId, saved, read) => {
+  if (saved) {
+    chrome.browserAction.setIcon({
+      path: { 16: icon16Saved, 32: icon32Saved, 128: icon128Saved },
+      tabId,
+    });
+  } else {
+    chrome.browserAction.setIcon({
+      path: { 16: icon16, 32: icon32, 128: icon128 },
+      tabId,
+    });
+  }
+
+  if (read) {
+    chrome.browserAction.setBadgeBackgroundColor({ color: '#ff4081', tabId });
+    chrome.browserAction.setBadgeText({ text: ' ', tabId });
+  } else {
+    chrome.browserAction.setBadgeText({ text: '', tabId });
+  }
+};
+
 export const tabHandler = (store) => (tabId, url) => {
-  store.dispatch(fetchArticleByUrl(url)).then(() => {
-    const state = store.getState();
-    const article = getArticle(state, url);
-    const user = getUser(state);
+  const state = store.getState();
+  const user = getUser(state);
 
-    if (isLoggedIn(user)) {
-      if (article.get('read')) {
-        chrome.browserAction.setBadgeBackgroundColor({ color: '#ff4081', tabId });
-        chrome.browserAction.setBadgeText({ text: ' ', tabId });
-      } else {
-        chrome.browserAction.setBadgeText({ text: '', tabId });
-      }
+  if (isLoggedIn(user)) {
+    setBadge(tabId, false, false);
```

Comment:
Why do we need to reset a badge before checking a status of url?
(it makes user experience worse since a badge is blinking)